tests/lib/rules/id-length.js

```diff
@@ -50,15 +51,20 @@ ruleTester.run("id-length", rule, {
         { code: "class Foo { method() {} }", parserOptions: { ecmaVersion: 6 } },
         { code: "function foo(...args) { }", parserOptions: { ecmaVersion: 6 } },
         { code: "var { prop } = {};", parserOptions: { ecmaVersion: 6 } },
-        { code: "var { prop: a } = {};", parserOptions: { ecmaVersion: 6 } },
+        { code: "var { a: foo } = {};", options: [{ min: 3 }], parserOptions: { ecmaVersion: 6 } },
+        { code: "var { prop: foo } = {};", options: [{ max: 3 }], parserOptions: { ecmaVersion: 6 } },
+        { code: "var { longName: foo } = {};", options: [{ min: 3, max: 5 }], parserOptions: { ecmaVersion: 6 } },
+        { code: "var { foo: a } = {};", options: [{ exceptions: ["a"] }], parserOptions: { ecmaVersion: 6 } },
         { code: "var { prop: [x] } = {};", parserOptions: { ecmaVersion: 6 } },
+        { code: "var { a: { b: { c: longName } } } = {};", parserOptions: { ecmaVersion: 6 } },
+        { code: "({ a: obj.x.y.z } = {});", options: [{ properties: "never" }], parserOptions: { ecmaVersion: 6 } },
+        { code: "var { a: [x]} = {};", parserOptions: { ecmaVersion: 6 } },
```

Comment:
> Also, I believe it will become invalid after #12839, so the build might fail when both PRs get merged.

@mdjermanovic 
Yes, right. it will conflict, and I'll resolve that Thanks 👍